### ISSUE TITLE
Distinction between Potentials and Calculators

### ISSUE THREAD

**Issue by cortner** [libAtoms/JuLIP.jl#6]
The distinction between calculators and Potentials looks more and more arbitrary. E.g., we have a `PairCalculator` which has as a single argument the pair potential.

Probably want to refactor code a little so that a Potential is a calculator and then use dispatch to choose the correct versions of energy, force, etc.  E.g. 
```julia
abstract PairPotential <: AbstractCalculator
type AnalyticPairPotential <: PairPotential  ... end 
function energy(pot::PairPotential, at::AbstractAtoms)
   # requires only that `pot` has expected pair potential behaviour
end
```

**Comment by jameskermode** [comment]
I agree that Potential should be a subtype of Calculator. Historically, everything in ASE is called a calculator and everythIng in QUIP is called a potential, which has caused some confusion for calculators which are not a (conservative) potential.

**Comment by cortner** [comment]
thank you - it is starting to feel more and more like the right thing to do.

**Comment by cortner** [comment]
Resolved via commit ce22ac7e756752b0184d76dbc67e6d5f2c5b3f27
